src/audio/copier/copier.c

```diff
@@ -2031,12 +2036,10 @@ static int copier_position(struct comp_dev *dev, struct sof_ipc_stream_posn *pos
 
 	switch (dev->ipc_config.type) {
 	case SOF_COMP_HOST:
+		/* only support host not support gtw case */
 		if (!cd->ipc_gtw) {
 			posn->host_posn = cd->hd->local_pos;
 			ret = posn->host_posn;
-		} else {
```

Comment:
you can maybe move it out of the switch and do if (cd->ipc_gtw) return 0. Then you can remove the check here.

Reply:
yes, however, from performance perspective, it may brings always extra cycles for "if", if is not small, normally, it brings 5 times MAC cycles, how about keep it?